cpp/src/arrow/python/numpy_convert.cc

```diff
@@ -384,6 +384,40 @@ Status SparseCSCMatrixToNdarray(const std::shared_ptr<SparseCSCMatrix>& sparse_t
   return SparseCSXMatrixToNdarray(sparse_tensor, base, out_data, out_indptr, out_indices);
 }
 
+Status SparseCSFTensorToNdarray(const std::shared_ptr<SparseCSFTensor>& sparse_tensor,
+                                PyObject* base, PyObject** out_data,
+                                PyObject** out_indptr, PyObject** out_indices) {
+  const auto& sparse_index = arrow::internal::checked_cast<const SparseCSFIndex&>(
+      *sparse_tensor->sparse_index());
+
+  // Wrap tensor data
+  OwnedRef result_data;
+  RETURN_NOT_OK(SparseTensorDataToNdarray(
+      *sparse_tensor, {sparse_tensor->non_zero_length(), 1}, base, result_data.ref()));
+
+  // Wrap indices
+  *out_indptr = PyList_New(0);
+  *out_indices = PyList_New(0);
+
+  for (const auto& indptr : sparse_index.indptr()) {
+    PyObject* item;
+    RETURN_NOT_OK(TensorToNdarray(indptr, base, &item));
+    if (PyList_Append(*out_indptr, item) < 0) {
+      RETURN_IF_PYERROR();
```

Comment:
Also need to `Py_CLEAR(*out_indptr)` and `Py_CLEAR(*out_indices)` before erroring out. You could probably use OwnedRef instead, to make things simpler.

Reply:
Made changes please check.

Reply:
Got it. Thanks!